### PR TITLE
chore(app-shell): migrate usb detection

### PR DESCRIPTION
**Pull request by brenthagen**
since `usb-detection` is [now deprecated](https://github.com/MadLittleMods/node-usb-detection), an initial attempt at migration to `usb`.

TODO:
 - adjust `UsbDevice` type
 - make necessary realtek adapter detection changes

### CODE REVIEW

File: app-shell-odd/Makefile [hunk 1]
```diff
@@ -65,7 +65,7 @@ package-deps: clean lib deps
 
 .PHONY: dist-ot3
 dist-ot3: package-deps
-	NO_USB_DETECTION=true OT_APP_DEPLOY_BUCKET=opentrons-app OT_APP_DEPLOY_FOLDER=builds OPENTRONS_PROJECT=ot3 $(builder) --linux --arm64
```
Comment (sfoster1) [inline app-shell-odd/Makefile:68]: we almost certainly still need to hold this out of ODD builds. it's a native node extension that forms a thing wrapper around libusb, but it still will have the problems with cross-compilation that usb-notifications had.

**Review by sfoster1 — CHANGES_REQUESTED** [review]
We should continue to hold this out of ODD builds - we'll have the same problem with cross compilation, and ODD just doesn't need it at all.

File: app-shell-odd/Makefile (hunk 1 above)
Comment (sfoster1) [reply app-shell-odd/Makefile:68]: Or is this because it's unnecessary since there was never an `app-shell-odd` dep on `usb-detection` in the first place?

Comment (brenthagen) [reply app-shell-odd/Makefile:68]: yes exactly, i think it's unnecessary since `app-shell-odd` doesn't have the dep

**Review by sfoster1 — APPROVED** [review]
Never mind, didn't understand the app-shell-odd situation

**Pull request closed without merging**